File: iprep/Models/QuizQuestion.swift
import Foundation

struct QuizOption: Identifiable, Hashable {
    let id: String
    let text: String
}

struct QuizQuestion: Identifiable, Hashable {
    let id: String
    let number: Int
    let prompt: String
    let options: [QuizOption]
    let correctOptionId: String
    let explanation: String
    let references: [String]
    let imageURLs: [URL]

    func option(withId id: String) -> QuizOption? {
        options.first { $0.id == id }
    }
}

struct QuizModule: Identifiable, Hashable {
    let summary: Module
    let questions: [QuizQuestion]

    var id: String { summary.id }
}

struct QuizSessionQuestion: Identifiable, Hashable {
    let module: QuizModule
    let question: QuizQuestion

    var id: String { question.id }
}

File: shared/src/commonMain/kotlin/com/russellmiller/iprep/shared/models/QuizQuestion.kt
package com.russellmiller.iprep.shared.models

import kotlinx.serialization.Serializable

@Serializable
data class QuizOption(
    val id: String,
    val text: String,
    val imageURL: String? = null
)

@Serializable
data class QuizQuestion(
    val id: String,
    val number: Int,
    val prompt: String,
    val options: List<QuizOption>,
    val correctOptionId: String,
    val explanation: String,
    val references: List<String>,
    val imageURLs: List<String>,
    val explanationImageURLs: List<String>
) {
    fun option(withId: String): QuizOption? {
        return options.firstOrNull { it.id == withId }
    }
}

@Serializable
data class QuizModule(
    val summary: Module,
    val questions: List<QuizQuestion>
) {
    val id: String get() = summary.id
}

@Serializable
data class QuizSessionQuestion(
    val module: QuizModule,
    val question: QuizQuestion
) {
    val id: String get() = question.id
}
